Codeinwp/neve
# fix: menu on legacy skin #3750 #3752

### Summary

Fix menu issues on the legacy skin.
Contains changes from #2935 and #3731


### Will affect the visual aspect of the product

YES

### Screenshots 

### Test instructions

1. Check the menu with the legacy skin
2. Check that the drop-down works and that the caret is displayed inline
3. Check the functionality related to #2935

When testing please use a replicate version that has the legacy and the development build as there might be things that are not working as expected on legacy prior to this PR and are not tied to these changes.

Let's keep the scope to only the changes for the keyboard navigation.


### Time Worked:
~ 9h
~ 20 min


Closes #3750.
Closes #3752.

Closes #2935.
Closes #3731.
